### PR TITLE
Add Enable Partitioning flag to the Topic Properties Constructor #17077

### PR DESCRIPTION
**Pull request by darrenhull**

### CODE REVIEW

**Comment by jsquire** [comment]
Hi @darrenhull.  Thank you for your interest in helping to improve the Azure SDK experience and for your contribution.    I'm going to defer to @JoshLove-msft as the authoritative voice on the proposed changes;   I've added some tags and routed this over to Josh to get things started.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs [hunk 1]
```diff
@@ -43,6 +43,7 @@ internal TopicProperties(CreateTopicOptions options)
             AuthorizationRules = options.AuthorizationRules;
             Status = options.Status;
             EnableBatchedOperations = options.EnableBatchedOperations;
+            EnablePartitioning = options.EnablePartitioning;
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs:46]: We'll also need to reflect this in the [Model Factory](https://github.com/Azure/azure-sdk-for-net/blob/d2bcf77a2b1ae1af07a915e751af4c68edd196e1/sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusModelFactory.cs#L151) and [Topic Properties Tests](https://github.com/Azure/azure-sdk-for-net/blob/d2bcf77a2b1ae1af07a915e751af4c68edd196e1/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Administration/TopicPropertiesTests.cs).    

Though in the latter, it appears that there is coverage only for the factory.  @JoshLove-msft: is that intentional or should we also have a test to ensure that the options are correctly implied?

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs:46]: Yep, good catch. I will go ahead and update these in a follow up PR. Thanks for the contribution @darrenhull.

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs (hunk 1 above)
Comment (darrenhull) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs:46]: Great, I’m happy to also follow up with a PR to update the Model Factory and the Topic Properties tests. I will create it on Monday. Just out of interest, when would the next nugget package be published? Just wondering if I should use my fork until it is.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/TopicProperties.cs:46]: Thanks @darrenhull we are on a pretty tight timeline for our next release so I will probably create the PR today. We expect the next nuget package to be published next week.

**Pull request merged**